### ISSUE TITLE
adding option for choosing Close() or Unsubscribe() on Disposing IStanSubscription

### ISSUE THREAD

**Issue by itn3000** [nats-io/csharp-nats-streaming#48]
I wrote the following code.
```csharp

namespace StanTest
{
    using System;
    using STAN.Client;
    using System.Threading.Tasks;
    using System.Threading;
    using System.Text;
    static class Tests
    {
        public static async Task CreateProducerTask(StanConnectionFactory cf, StanOptions opts)
        {
            using (var c = cf.CreateConnection("test-cluster", "stantest-producer", opts))
            {
                for (int i = 0; i < 1000; i++)
                {
                    await c.PublishAsync("foo", Encoding.UTF8.GetBytes($"x{i}")).ConfigureAwait(false);
                    await Task.Delay(10).ConfigureAwait(false);
                }
            }
            Console.WriteLine($"producer done");
        }
        public static async Task CreateConsumerTask(StanConnectionFactory cf, StanOptions opts, CancellationToken ct)
        {
            var subopts = StanSubscriptionOptions.GetDefaultOptions();
            // set DurableName for resuming.
            subopts.DurableName = "standurable";
            while (!ct.IsCancellationRequested)
            {
                using (var c = cf.CreateConnection("test-cluster", "stantest-consumer", opts))
                {
                    using (var sub = c.Subscribe("foo", subopts, (sender, ev) =>
                    {
                        Console.WriteLine($"recv:{Encoding.UTF8.GetString(ev.Message.Data)}");
                    }))
                    {
                        await Task.Delay(1000).ConfigureAwait(false);
                        // sub.Close();
                    }
                }
                // simulate reboot service
                Console.WriteLine($"rebooting subscription");
                await Task.Delay(1000).ConfigureAwait(false);
                Console.WriteLine($"rebooting subscription done");
            }
        }
        public static async Task DoTest(string natsUrl)
        {
            var cf = new StanConnectionFactory();
            var opts = StanOptions.GetDefaultOptions();
            opts.NatsURL = natsUrl;
            using (var cts = new CancellationTokenSource())
            {
                await Task.WhenAll(
                    Task.Run(async () =>
                    {
                        await CreateProducerTask(cf, opts).ConfigureAwait(false);
                        cts.Cancel();
                    })
                    ,
                    CreateConsumerTask(cf, opts, cts.Token)
                ).ConfigureAwait(false);
            }
        }
    }
}
```
I expected to output string "x[number]" continuously.
but some messages sent while closing subscriptions cannot be received by subscription.
This can be avoided by executing Close() before Dispose().(by surrounding try-finally)
So I suggest the option for choosing Close() or Unsubscribe() when disposing subscription.

**Comment by ColinSullivan1** [comment]
@itn3000 , Thank you for using NATS!

That sounds reasonable to me - dictating disposal behavior through an option would be a nice feature with respect to API usage.  I'll get this in by the next release, or a PR is always welcome.

Thanks,
Colin

**Comment by itn3000** [comment]
I created PR (#49)

**Comment by ColinSullivan1** [comment]
Resolved by #49.
